apps/gustaf/gustaf_matepairs.h

```diff
@@ -99,15 +99,28 @@ _importSequences(CharString const & fileNameL,
         append(qual, qualR);
         appendValue(seqs, seq, Generous());
         appendValue(quals, qual, Generous());
-        appendValue(ids, id, Generous());
+       // appendValue(ids, id, Generous());
 
         _getShortId(sId, id);
-        if (!_checkUniqueId(sId, id, ids, sIds))
-            ++counter;
-        appendValue(sIds, sId);
+        appendValue(ids, sId, Generous());
         clear(seq);
         clear(qual);
     }
+    
+    // Check for dupliacte id entries.
+    StringSet<TId> uniqueIds = ids;
+    std::sort(begin(uniqueIds, Standard()), end(uniqueIds, Standard()), IdComparator<TId>());  // O(n*log(n))
+    TIdSetIterator itOldEnd = end(uniqueIds, Standard());
+    TIdSetIterator itNewEnd = std::unique(begin(uniqueIds, Standard()), end(uniqueIds, Standard()), IdComparator<TId>());  // O(n)
+
+    --itNewEnd;
```

Comment:
Why do you decrease it?
I thought itNewEnd points behind the last element not removed which is the same as itOldEnd if nothing was removed.